Sources/Display.php

```diff
@@ -841,37 +823,84 @@ function Display()
 		call_integration_hook('integrate_poll_buttons');
 	}
+	
+	if(!empty($_REQUEST['page_id']))
+		$start_char = substr($_REQUEST['page_id'], 0, 1);
+	else
+		$start_char = null;
+
+	if ($start_char === 'M' || $start_char === 'L')
+		$page_id = substr($_REQUEST['page_id'], 1);
 
-	// Calculate the fastest way to get the messages!
-	$ascending = empty($options['view_newest_first']);
-	$start = $_REQUEST['start'];
 	$limit = $context['messages_per_page'];
-	$firstIndex = 0;
-	if ($start >= $context['total_visible_posts'] / 2 && $context['messages_per_page'] != -1)
+ 
+	// Jump to page
+	if (empty($start_char))
 	{
-		$ascending = !$ascending;
-		$limit = $context['total_visible_posts'] <= $start + $limit ? $context['total_visible_posts'] - $start : $limit;
-		$start = $context['total_visible_posts'] <= $start + $limit ? 0 : $context['total_visible_posts'] - $start - $limit;
-		$firstIndex = $limit - 1;
-	}
+		// Calculate the fastest way to get the messages!
+		$ascending = empty($options['view_newest_first']);
+		$start = $_REQUEST['start'];
+		$firstIndex = 0;
+		if ($start >= $context['total_visible_posts'] / 2 && $context['messages_per_page'] != -1)
+		{
+			$ascending = !$ascending;
+			$limit = $context['total_visible_posts'] <= $start + $limit ? $context['total_visible_posts'] - $start : $limit;
+			$start = $context['total_visible_posts'] <= $start + $limit ? 0 : $context['total_visible_posts'] - $start - $limit;
+			$firstIndex = $limit - 1;
+		}
 
-	// Get each post and poster in this topic.
-	$request = $smcFunc['db_query']('', '
-		SELECT id_msg, id_member, approved
-		FROM {db_prefix}messages
-		WHERE id_topic = {int:current_topic}' . (!$modSettings['postmod_active'] || $approve_posts ? '' : '
-		AND (approved = {int:is_approved}' . ($user_info['is_guest'] ? '' : ' OR id_member = {int:current_member}') . ')') . '
-		ORDER BY id_msg ' . ($ascending ? '' : 'DESC') . ($context['messages_per_page'] == -1 ? '' : '
-		LIMIT {int:start}, {int:max}'),
-		array(
-			'current_member' => $user_info['id'],
-			'current_topic' => $topic,
-			'is_approved' => 1,
-			'blank_id_member' => 0,
-			'start' => $start,
-			'max' => $limit,
-		)
-	);
+		// Get each post and poster in this topic.
+		$request = $smcFunc['db_query']('', '
+			SELECT id_msg, id_member, approved
+			FROM {db_prefix}messages
+			WHERE id_topic = {int:current_topic}' . (!$modSettings['postmod_active'] || $approve_posts ? '' : '
+			AND (approved = {int:is_approved}' . ($user_info['is_guest'] ? '' : ' OR id_member = {int:current_member}') . ')') . '
+			ORDER BY id_msg ' . ($ascending ? '' : 'DESC') . ($context['messages_per_page'] == -1 ? '' : '
+			LIMIT {int:start}, {int:max}'),
+			array(
+				'current_member' => $user_info['id'],
+				'current_topic' => $topic,
+				'is_approved' => 1,
+				'blank_id_member' => 0,
+				'start' => $start,
+				'max' => $limit,
+			)
+		);
+	}
+	else //next or before page
+	{
+		$firstIndex = 0;
+		
+		if ($start_char === 'M')
```

Comment:
in fact $ascending can be simplified as 
```php
$ascending = $start_char === 'M';
```
no need for the explicit true and false there.